docs/running-rules.md

```diff
@@ -35,6 +35,8 @@ $ semgrep -e '$X == $X' --lang=py path/to/src
 $ semgrep --config path/to/yaml
 ```
 
+We are working on optimizations to improve Semgrep's performance, which may have unexpected behavior. If you encounter a bug, you can run with `--optimizations none` to use the original code path. 
```

Comment:
@emjin Could this be more specific about the changes and the bugs that people may encounter, or how those bugs may present to the user?

One suggestion:
“Changes to the way Semgrep pre-processes rules introduced in Semgrep v0.55.0 include optimizations that improve speed but may sometimes lead to unexpected behavior. If you encounter {X,Y,Z} issues, try running Semgrep with `--optimizations none`.